backend/core/src/main/java/org/sonarsource/sonarlint/core/sca/DependencyRiskService.java

```diff
@@ -202,33 +203,40 @@ private static ServerDependencyRisk.Transition adaptTransition(DependencyRiskTra
-      .map(pkg -> {
-        var recommendationDetails = pkg.recommendationDetails();
-        RecommendationDetailsDto recommendationDetailsDto = null;
-        if (recommendationDetails != null) {
-          recommendationDetailsDto = RecommendationDetailsDto.builder()
-            .impactScore(recommendationDetails.impactScore())
-            .impactDescription(recommendationDetails.impactDescription())
-            .realIssue(recommendationDetails.realIssue())
-            .falsePositiveReason(recommendationDetails.falsePositiveReason())
-            .includesDev(recommendationDetails.includesDev())
-            .specificMethodsAffected(recommendationDetails.specificMethodsAffected())
-            .specificMethodsDescription(recommendationDetails.specificMethodsDescription())
-            .otherConditions(recommendationDetails.otherConditions())
-            .otherConditionsDescription(recommendationDetails.otherConditionsDescription())
-            .workaroundAvailable(recommendationDetails.workaroundAvailable())
-            .workaroundDescription(recommendationDetails.workaroundDescription())
-            .visibility(recommendationDetails.visibility()).build();
-        }
-        return new AffectedPackageDto(pkg.purl(), pkg.recommendation(), recommendationDetailsDto);
-      })
-      .toList();
+    var affectedPackages = new ArrayList<AffectedPackageDto>();
+    String vulnerabilityId = null;
+    String description = null;
+    if (serverResponse.vulnerability() != null) {
+      serverResponse.vulnerability().affectedPackages()
+        .forEach(pkg -> {
+          var recommendationDetails = pkg.recommendationDetails();
+          RecommendationDetailsDto recommendationDetailsDto = null;
+          if (recommendationDetails != null) {
+            recommendationDetailsDto = RecommendationDetailsDto.builder()
+              .impactScore(recommendationDetails.impactScore())
+              .impactDescription(recommendationDetails.impactDescription())
+              .realIssue(recommendationDetails.realIssue())
+              .falsePositiveReason(recommendationDetails.falsePositiveReason())
+              .includesDev(recommendationDetails.includesDev())
+              .specificMethodsAffected(recommendationDetails.specificMethodsAffected())
+              .specificMethodsDescription(recommendationDetails.specificMethodsDescription())
+              .otherConditions(recommendationDetails.otherConditions())
+              .otherConditionsDescription(recommendationDetails.otherConditionsDescription())
+              .workaroundAvailable(recommendationDetails.workaroundAvailable())
+              .workaroundDescription(recommendationDetails.workaroundDescription())
+              .visibility(recommendationDetails.visibility()).build();
+          }
+          affectedPackages.add(new AffectedPackageDto(pkg.purl(), pkg.recommendation(), recommendationDetailsDto));
```

Comment:
Nitpick: I think you could still use a  `map` instead of having another collection:

`var affectedPackages = vulnerability == null ? List.of() : vulnerability().affectedPackages().stream().map(.....`